File: TypeScript/longest_common_prefix/14.test.ts
import longestCommonPrefix from './14';
import { describe, test, expect } from 'bun:test';

interface tt {
  strs: string[];
  want: string;
}

const tests: tt[] = [
  {
    strs: ['flower', 'flow', 'flight'],
    want: 'fl',
  },
  {
    strs: ['dog', 'racecar', 'car'],
    want: '',
  },
];

describe('longest common prefix', () => {
  tests.forEach((t, i) => {
    test(`case ${i + 1}`, () => {
      const result = longestCommonPrefix(t.strs);
      expect(result).toBe(t.want);
    });
  });
});

File: Golang/longest_common_prefix/14_test.go
package longest_common_prefix

import (
	"fmt"
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestLongestCommonPrefix(t *testing.T) {
	cases := []struct {
		strs []string
		want string
	}{
		{strs: []string{"flower", "flow", "flight"}, want: "fl"},
		{strs: []string{"dog", "racecar", "car"}, want: ""},
		{strs: []string{"chi", "chien", "chau"}, want: "ch"},
		{strs: []string{"", "", ""}, want: ""},
		{strs: []string{"apple", "apricot", "ape"}, want: "ap"},
		{strs: []string{"", "123", "ab"}, want: ""},
		{strs: []string{"race", "racer", "races"}, want: "race"},
	}

	for i, tc := range cases {
		t.Run(fmt.Sprintf("case %d", i+1), func(t *testing.T) {
			result := longestCommonPrefix(tc.strs)
			assert.Equal(t, tc.want, result, fmt.Sprintf("test case %d failed", i))
		})
	}
}
